Gems/Atom/RHI/Code/Include/Atom/RHI/ShaderResourceGroupData.h

```diff
@@ -12,6 +12,16 @@
 #include <Atom/RHI/ImageView.h>
 #include <Atom/RHI/Buffer.h>
 #include <Atom/RHI/BufferView.h>
+#include <AzCore/Preprocessor/Enum.h>
+
+AZ_ENUM_CLASS_WITH_UNDERLYING_TYPE(BindlessResourceType, uint32_t,
+            m_Texture2D,                     //ReadTexture
+            m_RWTexture2D,                   //ReadWriteTexture
+            m_TextureCube,                   //ReadTextureCube
+            m_ByteAddressBuffer,             //ReadBuffer
+            m_RWByteAddressBuffer,           //ReadWriteBuffer
+            Count
+        );
```

Comment:
did you mean to move this outside the `AZ::RHI` namespace?

Reply:
Done